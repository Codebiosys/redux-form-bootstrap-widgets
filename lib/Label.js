'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _reactBootstrap = require('react-bootstrap');

require('bootstrap/dist/css/bootstrap.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (_ref) {
  var label = _ref.label,
      required = _ref.required;

  if (label && required) {
    return React.createElement(
      _reactBootstrap.ControlLabel,
      null,
      label,
      '\xA0',
      React.createElement(
        'small',
        null,
        React.createElement(_reactBootstrap.Glyphicon, { glyph: 'asterisk', style: { color: '#a94442' } })
      )
    );
  } else if (label) {
    return React.createElement(
      _reactBootstrap.ControlLabel,
      null,
      label
    );
  }
  return null;
};