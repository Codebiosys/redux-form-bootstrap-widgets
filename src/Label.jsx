import React from 'react';
import PropTypes from 'prop-types';

import { ControlLabel, Glyphicon } from 'react-bootstrap';

import 'bootstrap/dist/css/bootstrap.css';

const Label = ({ label, required }) => {
  if (label && required) {
    return (<ControlLabel>{label}&nbsp;<small><Glyphicon glyph="asterisk" style={{ color: '#a94442' }} /></small></ControlLabel>);
  } else if (label) {
    return (<ControlLabel>{label}</ControlLabel>);
  }
  return null;
};

Label.propTypes = {
  label: PropTypes.string,
  required: PropTypes.bool,
};

Label.defaultProps = {
  label: '',
  required: false,
};

export default Label;
