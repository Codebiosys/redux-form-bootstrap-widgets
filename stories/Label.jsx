import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Label from '../src/Label';


storiesOf('Label', module)
  .add('with text', () => (
    <Label label="hello">Hello Button</Label>
  ))
  .add('with required', () => (
    <Label label="hello" required>Hello Button</Label>
  ));
