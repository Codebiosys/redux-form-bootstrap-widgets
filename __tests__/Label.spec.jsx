/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import Label from 'Label';

const fieldProps = {
  label: 'Field Label',
  required: false,
};

const requiredFieldProps = {
  label: 'Field Label',
  required: true,
};

describe('The Label', () => {
  let inputWrapper;

  beforeEach(() => {
    inputWrapper = shallow(<Label {...fieldProps} />);
  });

  it('renders', () => {
    expect(inputWrapper).toMatchSnapshot();
  });

  it('has no asterisk when not required', () => {
    expect(inputWrapper.find('Glyphicon').exists()).toBe(false);
  });

  it('has an asterisk when required', () => {
    const requiredWrapper = shallow(<Label {...requiredFieldProps} />);
    expect(requiredWrapper.find('Glyphicon').prop('glyph')).toEqual('asterisk');
  });

  it('is empty when there is no label', () => {
    const emptywrapper = shallow(<Label />);
    expect(emptywrapper.getElement()).toBe(null);
  });
});
