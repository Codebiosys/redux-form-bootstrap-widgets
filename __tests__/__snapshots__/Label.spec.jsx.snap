// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Field Set renders 1`] = `
<ControlLabel
  bsClass="control-label"
  srOnly={false}
>
  Field Label
</ControlLabel>
`;
